index.js

```diff
@@ -231,47 +230,46 @@ class GithubScm extends Scm {
 
     /**
      * Adds the Screwdriver webhook to the Github repository
-     * @method _addWebhook
-     * @param  {Object}    config            Config object
-     * @param  {String}    config.scmUri     The SCM URI to add the webhook to
-     * @param  {String}    config.token      Service token to authenticate with Github
-     * @param  {String}    config.webhookUrl The URL to use for the webhook notifications
-     * @return {Promise}                     Resolve means operation completed without failure.
+     * @async  _addWebhook
+     * @param  {Object}    config             Config object
+     * @param  {String}    config.scmUri      The SCM URI to add the webhook to
+     * @param  {String}    config.token       Service token to authenticate with Github
+     * @param  {String}    config.webhookUrl  The URL to use for the webhook notifications
+     * @return {Promise}                      Resolve means operation completed without failure
      */
-    _addWebhook(config) {
-        return this.lookupScmUri({
+    async _addWebhook(config) {
+        const scmInfo = await this.lookupScmUri({
             scmUri: config.scmUri,
             token: config.token
-        }).then(scmInfo =>
-            this._findWebhook({
-                scmInfo,
-                url: config.webhookUrl,
-                page: 1,
-                token: config.token
-            }).then(hookInfo =>
-                this._createWebhook({
-                    hookInfo,
-                    scmInfo,
-                    token: config.token,
-                    url: config.webhookUrl
-                })
-            )
-        );
+        });
+        const hookInfo = await this._findWebhook({
+            scmInfo,
+            url: config.webhookUrl,
+            page: 1,
+            token: config.token
+        });
+
+        return this._createWebhook({
+            hookInfo,
+            scmInfo,
+            token: config.token,
+            url: config.webhookUrl
+        });
     }
 
     /**
-     * Checkout the source code from a repository; resolves as an object with checkout commands
-     * @method getCheckoutCommand
+     * Get checkout the source code from a repository
+     * @async  getCheckoutCommand
      * @param  {Object}    config
      * @param  {String}    config.branch        Pipeline branch
      * @param  {String}    config.host          Scm host to checkout source code from
      * @param  {String}    config.org           Scm org name
      * @param  {String}    config.repo          Scm repo name
      * @param  {String}    config.sha           Commit sha
      * @param  {String}    [config.prRef]       PR reference (can be a PR branch or reference)
-     * @return {Promise}
+     * @return {Promise}                        Resolves to object containing name and checkout commands
      */
-    _getCheckoutCommand(config) {
+    async _getCheckoutCommand(config) {
```

Comment:
Note: I use the `async` keyword here so I can do

```js
return {
  name: 'sd-checkout-code',
  command: command.join(' && ')
};
```

Instead of

```js
return Promise.resolve({
  name: 'sd-checkout-code',
  command: command.join(' && ')
});
```